RoddyCore/src/de/dkfz/roddy/core/ProjectLoader.groovy

```diff
@@ -169,6 +192,56 @@ class ProjectLoader {
             throw new ProjectLoaderException("Could not extract analysis for ${configurationIdentifier}.");
     }
 
+    void checkAndPreparePremisesOrFail() {
+        if (projectSource == PSRC_UNSET)
+            throw new ProjectLoaderException("There is no project identifier set. It is not possible to run without setting projectSource in the ProjectLoader.")
+
+        if (projectSource == PSRC_CONFIGURATION)
+            return
+
+        if (projectSource == PSRC_PLUGIN) {
+            // Create a project configuration, if all premises are valid.
+
+            // Check, if the i/o directory is set on the command line
+            if (!Roddy.useCustomIODirectories())
+                throw new ProjectLoaderException("It is not possible to use the project configuration free mode without the --${RoddyStartupOptions.useiodir.name()} option.")
+
+            // Create a temporary project configuration in ~/.roddy/configurationFreeMode
+            String configurationFileName = "projects" + projectID + "_" + analysisID + "_" + Integer.toHexString(Roddy.commandLineCall.getOptionValue(RoddyStartupOptions.useiodir).hashCode()) + ".sh"
+
+            String newProjectID = "CFreeMode_" + Integer.toHexString(configurationFileName.hashCode())
+            String pluginID = projectID.replaceFirst(StringConstants.SPLIT_UNDERSCORE, StringConstants.COLON)
+            String analysisImport = analysisID + "," + (analysisID.endsWith("Analysis") ? analysisID : analysisID + "Analysis") + "," + pluginID
+            List<String> lines = []
+            lines << "#name " + newProjectID
+            lines << "#analysis " + analysisImport
+            if (Roddy.isOptionSet(RoddyStartupOptions.baseconfig)) lines << "#imports " + Roddy.commandLineCall.getOptionValue(RoddyStartupOptions.baseconfig)
+
+            lines << "inputBaseDirectory=" + Roddy.customBaseInputDirectory
+            lines << "outputBaseDirectory=" + Roddy.customBaseOutputDirectory
+            lines << 'outputAnalysisBaseDirectory=${outputBaseDirectory}/${pid}'
+
+            projectID = "CFreeMode_" + Integer.toHexString(configurationFileName.hashCode())
+
+            File configurationFile = new File(Roddy.getFolderForConfigurationFreeMode(), configurationFileName)
+
+            logger.always("Will use automatically generated configuration file: ${configurationFileName}")
+
+            String configurationText = lines.join("\n")
+            if (configurationFile.exists()) {
+                if (RoddyIOHelperMethods.getMD5OfText(configurationText) != RoddyIOHelperMethods.getMD5OfFile(configurationFile)) {
```

Comment:
Hm, I think this should not happen silently. Do we want to have a copy of the old file, or is a "copy" of the config in the output paths sufficient?

And why do you compare the MD5s instead of the texts? For the md5s the texts has to be traversed fully, for the comparison not. And with a plain comparison the code would get simpler.